gear-programs/bridging-payment-vara-supply/src/services/mod.rs

```diff
@@ -0,0 +1,370 @@
+        receiver: H160,
+        eth_token_id: H160,
+    },
+}
+
+static mut DATA: Option<BridgingPaymentData> = None;
+static mut CONFIG: Option<Config> = None;
+static mut MSG_TRACKER: Option<MessageTracker> = None;
+
+#[derive(Debug)]
+pub struct BridgingPaymentData {
+    admin_address: ActorId,
+    vft_treasury_address: ActorId,
+}
+
+#[derive(Debug, Decode, Encode, TypeInfo)]
+pub struct InitConfig {
+    admin_address: ActorId,
+    vft_treasury_address: ActorId,
+    config: Config,
+}
+
+impl InitConfig {
+    pub fn new(admin_address: ActorId, vft_treasury_address: ActorId, config: Config) -> Self {
+        Self {
+            admin_address,
+            vft_treasury_address,
+            config,
+        }
+    }
+}
+
+#[derive(Debug, Decode, Encode, TypeInfo, Clone)]
+pub struct Config {
+    fee: u128,
+    gas_for_reply_deposit: u64,
+    gas_to_send_request_to_treasury: u64,
+    reply_timeout: u32,
+    gas_for_request_to_treasury_msg: u64,
+}
+
+impl Config {
+    pub fn new(
+        fee: u128,
+        gas_for_reply_deposit: u64,
+        gas_to_send_request_to_treasury: u64,
+        reply_timeout: u32,
+        gas_for_request_to_treasury_msg: u64,
+    ) -> Self {
+        Self {
+            fee,
+            gas_for_reply_deposit,
+            gas_to_send_request_to_treasury,
+            reply_timeout,
+            gas_for_request_to_treasury_msg,
+        }
+    }
+}
+impl<T> BridgingPayment<T>
+where
+    T: ExecContext,
+{
+    pub fn seed(config: InitConfig) {
+        unsafe {
+            DATA = Some(BridgingPaymentData {
+                admin_address: config.admin_address,
+                vft_treasury_address: config.vft_treasury_address,
+            });
+            CONFIG = Some(config.config);
+            MSG_TRACKER = Some(MessageTracker::default());
+        }
+    }
+    pub fn new(exec_context: T) -> Self {
+        Self { exec_context }
+    }
+
+    fn data(&self) -> &BridgingPaymentData {
+        unsafe {
+            DATA.as_ref()
+                .expect("BridgingPaymentData::seed() should be called")
+        }
+    }
+
+    fn data_mut(&mut self) -> &mut BridgingPaymentData {
+        unsafe {
+            DATA.as_mut()
+                .expect("BridgingPaymentData::seed() should be called")
+        }
+    }
+
+    fn config(&self) -> &Config {
+        unsafe {
+            CONFIG
+                .as_ref()
+                .expect("BridgingPaymentData::seed() should be called")
+        }
+    }
+
+    fn config_mut(&mut self) -> &mut Config {
+        unsafe {
+            CONFIG
+                .as_mut()
+                .expect("BridgingPaymentData::seed() should be called")
+        }
+    }
+}
+
+#[service(events = BridgingPaymentEvents)]
+impl<T> BridgingPayment<T>
+where
+    T: ExecContext,
+{
+    pub fn set_fee(&mut self, fee: u128) {
+        let data = self.data();
+        if data.admin_address != self.exec_context.actor_id() {
+            panic!("Not admin");
+        }
```

Comment:
I suggest moving it to separate fn `ensure_admin` and use it in all methods that require admin access

Reply:
Fixed